packages/model-viewer/src/three-components/ARRenderer.ts

```diff
@@ -383,53 +383,44 @@ export class ARRenderer extends EventDispatcher {
   }
 
   private updateView(view: XRView) {
-    const viewMatrix = view.transform.matrix;
-
     const scene = this.presentedScene!;
-    const {camera} = scene;
-    camera.near = 0.1;
-    camera.far = 100;
 
     (this.threeRenderer.xr as any).updateCamera(camera);
+    //@ts-ignore
+    const xrCamera = this.threeRenderer.xr.getCamera();
+
+    xrCamera.position.copy(camera.position);
+    xrCamera.quaternion.copy(camera.quaternion);
+    xrCamera.scale.copy(camera.scale);
```

Comment:
@mrdoob This all worked out pretty nicely, except that the decompose function is only run on the camera that's input to `updateCamera` rather than on the vrCamera itself. Would be a simple fix to make using XRManager a bit cleaner.

Reply:
Oh. Interesting...

Reply:
https://github.com/mrdoob/three.js/pull/21964